src/factor_nodes/addition.jl

```diff
@@ -50,4 +50,24 @@ function +(in1::Variable, in2)
     in1 + in2
 end
 
-+(in1, in2::Variable) = +(in2, in1)
\ No newline at end of file
++(in1, in2::Variable) = +(in2, in1)
+
+
+"""
+    -(in1::Variable, in2::Variable)
+
+A subtraction constraint based on the addition factor node.
+
+"""
+function -(in1::Variable, in2::Variable)
+    out = Variable()
+    Addition(in1, out, in2)
+    return out
+end
+
+function -(in1::Variable, in2)
+    @ensureVariables(in2)
+    in1 - in2
+end
+
+-(in1, in2::Variable) = -(in2, in1)
```

Comment:
Note that the `-` operator does not commute like the `+` operator, this line will lead to a faulty specification. Instead, consider
```suggestion
function -(in1, in2::Variable)
    @ensureVariables(in1)
    in1 - in2
end
```
In order to prevent confusion, it would also be good to write a test that verifies whether the interfaces are connected as expected.